ports/esp32/mphalport.c

```diff
@@ -75,7 +81,16 @@ void check_esp_err(esp_err_t code) {
             return;
         }
         o_str->base.type = &mp_type_str;
+        #if MICROPY_ERROR_REPORTING > MICROPY_ERROR_REPORTING_NORMAL
+        char err_msg[64];
+        esp_err_to_name_r(code, err_msg, sizeof(err_msg));
+        vstr_t vstr;
+        vstr_init(&vstr, 256);
+        vstr_printf(&vstr, "0x%04X %s in function '%s' at line %d in file '%s'", code, err_msg, func, line, file);
+        o_str->data = (const byte *)vstr.buf;
```

Comment:
You need to use `vstr_null_terminated` to access the buffer and make sure it's null terminated.